### PR TITLE
:arrow_up:  upgrade react-modal to v3.16.1 and remove the patch

### PR DESCRIPTION
**Pull request by MatissJanis**
Upgrading `react-modal` and removing the mockey patch.

**Is the removal of the monkey-path safe?**
I think so.. but I'm not 100% sure. And the git history doesn't help either. So I propose to remove it if we feel reasonably confident that things still work as expected. And if we notice issues after releasing: we can always roll back the change. However: at that point we will actually have some more context on the patch. Thus - win-win.

**What did the refactor of `ScopeTab` change?**

Nothing, it was replaced with react-aria/focus.

The implementation focuses on this part of the UI. It forces the focus to stay within the popover when it's open. So you can "tab" as many times as you want and it will remain in the same popover.

Screenshot 2023-03-01 at 20 54 30

---

Before:
```
File sizes after gzip:

  404.26 kB  build/static/js/main.304b19ba.js
  46.97 kB   build/static/js/52.d0589b8d.chunk.js
  5.35 kB    build/static/js/969.17029516.chunk.js
  3.05 kB    build/static/js/145.69afa4e0.chunk.js
  2.89 kB    build/static/css/main.5456b6c5.css
  389 B      build/static/media/browser-server.38078d2847e4787cc481.js
```

After:
```
File sizes after gzip:

  407.38 kB  build/static/js/main.cc5a5564.js
  46.97 kB   build/static/js/52.d0589b8d.chunk.js
  5.35 kB    build/static/js/969.17029516.chunk.js
  3.05 kB    build/static/js/145.69afa4e0.chunk.js
  2.89 kB    build/static/css/main.5456b6c5.css
  389 B      build/static/media/browser-server.38078d2847e4787cc481.js
```

### CODE REVIEW

**Comment by j-f1** [comment]
One small regression I noticed is that using option+tab in Safari worked properly with the old code but doesn’t work right anymore (focus moves correctly to the “Apply” button but then toggles between the address bar and the button instead of wrapping around).

File: packages/desktop-client/src/components/accounts/Filters.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import React, { useState, useRef, useEffect, useReducer } from 'react';
 import { useSelector } from 'react-redux';
 
+import { FocusScope } from '@react-aria/focus';
```
Comment (trevdor) [inline packages/desktop-client/src/components/accounts/Filters.js:4]: 🥇

**Review by trevdor — APPROVED** [review]
New focus trap worked well in my testing.

**Comment by MatissJanis** [comment]
> One small regression I noticed is that using option+tab (forces tabbing through all interactive elements rather than just `<input>`/`<select>`) in Safari worked properly with the old code but doesn’t work right anymore (focus moves correctly to the “Apply” button but then toggles between the address bar and the button instead of wrapping around).

True and good catch. Do you think it's a blocker for this cleanup? My guess would be that nobody actually noticed this `focus trapping` feature before this PR was opened.

**Comment by j-f1** [comment]
I think it’s fine, I imagine that most people who use tab to navigate will have that checkbox checked.

**Review by j-f1 — APPROVED** [review]

**Pull request merged**